reference_guide/navigation/navigation_intro.md

```diff
@@ -0,0 +1,93 @@
+  [Copyright profiles](https://www.jetbrains.com/help/idea/copyright-profiles.html).
+* Going to locations within files based on [Bookmarks](https://www.jetbrains.com/help/idea/navigating-with-bookmarks.html),
+  [TODOs](https://www.jetbrains.com/help/idea/using-todo.html), 
+  or [Lines](https://www.jetbrains.com/help/idea/2017.3/navigating-to-line.html). 
+
+### The Navigatable Interface 
+The [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface defines the following methods of particular interest:
+
+```java
+
+ boolean canNavigate();
+
+ boolean canNavigateToSource();
+ 
+ void navigate(boolean requestFocus); 
+```
+The `canNavigate()` method indicates whether this object supports navigation of any kind. The `canNavigateToSource()` method
+indicates whether this object supports navigation to source within an editor. See the 
+[Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface comments for more information about these two methods.
+
+The `navigate()` method is called to open the (content-containing) object in an editor or display, and request focus. Typically an implementation
+of `Navigatable` does the heavy lifting of invoking an editor for the content and requesting focus. See below for examples,
+and the [Navigatable](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/pom/Navigatable.java)
+interface for more information.  
+
+### Example IntelliJ Platform Implementations of Navigatable
+This section describes some example implementations that can be found in the IntelliJ Platform. This is by no means an exhaustive list, 
+but is intended to illustrate how `Navigatable` works in different contexts.
+
+#### Example Navigatable Class Related to Psi Elements
+
+The [NavigatablePsiElement](https://github.com/JetBrains/intellij-community/blob/master/platform/core-api/src/com/intellij/psi/NavigatablePsiElement.java) 
+interface is the basis for many implementation classes, abstract classes, and interfaces relating to `Navigatable` 
+Psi elements. The class diagram below shows `NavigatablePsiElement` inherits from three interfaces defining the underlying
+functionality of a `Navigatable` Psi element.
+
+<img src="img/navigatable_psi_element.png" alt="NavigatablePsiElement" width="500px"/>
+
+For example, the subclass [JavaStubPsiElement](https://github.com/JetBrains/intellij-community/blob/master/java/java-psi-impl/src/com/intellij/psi/impl/source/JavaStubPsiElement.java) 
+implementation is the basis for a variety of Psi element classes that
+represent Java files, classes, methods, and fields.
+
+#### Example Navigatable Class Related to Files
+
+The [OpenFileDescriptor](https://github.com/JetBrains/intellij-community/blob/master/platform/platform-api/src/com/intellij/openapi/fileEditor/OpenFileDescriptor.java)
```

Comment:
`OpenFileDescriptor` is also not just an example implementation, it's one of the common APIs plugin developers use to trigger navigation. It allows a plugin developer to open a specific source location in the editor, by specifying the file, line and column. It also supports more complex use cases such as controlling navigation within a split editor. This section should describe how a plugin developer can create an `OpenFileDescriptor` instance and use it to trigger navigation.